cbcontainers/models/compatibility_matrix_test.go

```diff
@@ -0,0 +1,58 @@
+package models_test
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	"github.com/vmware/cbcontainers-operator/cbcontainers/models"
+)
+
+func TestCheckCompatibility(t *testing.T) {
+	testCases := []struct {
+		min           models.AgentVersion
+		max           models.AgentVersion
+		agent         string
+		areCompatible bool
+	}{
+		{min: models.AgentVersion("2.7"), max: models.AgentVersion("2.8"), agent: "2.7", areCompatible: true},
+		{min: models.AgentVersion("2.7"), max: models.AgentVersion("2.8"), agent: "2.8", areCompatible: true},
+		{min: models.AgentVersion("2.7"), max: models.AgentVersion("2.8"), agent: "2.7.1", areCompatible: true},
+		{min: models.AgentVersion("2.7"), max: models.AgentVersion("2.8"), agent: "2.7.5", areCompatible: true},
+		{min: models.AgentVersion("2.7"), max: models.AgentVersion("2.8.1"), agent: "2.8.1", areCompatible: true},
+		{min: models.AgentVersion("2.7"), max: models.AgentVersion("2.8.1"), agent: "2.7.1-alpha", areCompatible: true},
+		{min: models.AgentVersion("2.7"), max: models.AgentVersion("2.8.1"), agent: "2.7.1-beta", areCompatible: true},
+		{min: models.AgentMinVersionNone, max: models.AgentVersion("2.8.1"), agent: "0.0.1", areCompatible: true},
+		{min: models.AgentVersion("2.7"), max: models.AgentMaxVersionLatest, agent: "3.0.0", areCompatible: true},
+		{min: models.AgentVersion("2.7"), max: models.AgentMaxVersionLatest, agent: "50.0.0", areCompatible: true},
+		{min: models.AgentVersion("1.0"), max: models.AgentVersion("2.8"), agent: "1.5", areCompatible: true},
+		{min: models.AgentVersion("1.0"), max: models.AgentVersion("2.8"), agent: "2.0", areCompatible: true},
+		{min: models.AgentVersion("0.0"), max: models.AgentVersion("2.8"), agent: "0.0", areCompatible: true},
+		{min: models.AgentVersion("0.0"), max: models.AgentVersion("2.8"), agent: "1.0", areCompatible: true},
+		{min: models.AgentVersion("0.0"), max: models.AgentVersion("2.8"), agent: "2.0", areCompatible: true},
+
+		{min: models.AgentMinVersionNone, max: models.AgentMaxVersionLatest, agent: "0.0", areCompatible: true},
+		{min: models.AgentMinVersionNone, max: models.AgentMaxVersionLatest, agent: "0.0.1", areCompatible: true},
+		{min: models.AgentMinVersionNone, max: models.AgentMaxVersionLatest, agent: "0.0.0-beta", areCompatible: true},
+		{min: models.AgentMinVersionNone, max: models.AgentMaxVersionLatest, agent: "30.0.0-beta", areCompatible: true},
+
+		{min: models.AgentVersion("2.7"), max: models.AgentVersion("2.8"), agent: "2.8.1", areCompatible: false},
```

Comment:
An interesting test case here is 
`minVersion=2.7.0, agentVersion=2.7.0-beta`

Which requires that we decide whether a beta/alpha is < than the actual version. In the test cases above we treat X.Y-beta > X.(Y-1) but it is not clear whether X.Y-beta < X.Y 

I think this is a very edge cases and I am OK with ignoring it